sounddevice.py

```diff
@@ -2289,14 +2289,40 @@ def reset(self):
         vars(self).clear()
         self.__init__()
 
+
 if not hasattr(_ffi, 'I_AM_FAKE'):
     # This object shadows the 'default' class, except when building the docs.
     default = default()
 
 
 class PortAudioError(Exception):
-    """This exception will be raised on PortAudio errors."""
+    """This exception will be raised on PortAudio errors.
+
+    Parameters
+    ----------
+    error_message : str, optional
+        User error message.
+    pa_error_code : int, optional
+        PaErrorCode value.
+    hosterror_info : tuple, optional
+        3-tuple containing the host API index, host error code, and the host
+        error message.
+
+    """
+    def __str__(self):
+        errormsg = self.args[0] if len(self.args) > 0 else ''
+
+        if len(self.args) > 2:
+            host_api, hosterror_code, hosterror_text = self.args[2]
+            hostname = query_hostapis(host_api)['name']
+            errormsg += " [PaErrorCode {0}, {1} error {2}]"\
+                        .format(self.args[1], hostname, hosterror_code)
+            if hosterror_text:
+                errormsg += ": '{0}'".format(hosterror_text)
```

Comment:
In light of your 2 comments, I would actually like to have the formatting like this: 
```
PortAudioError: Error opening Stream: Unanticipated host error [PaErrorCode -9999]: 'Invalid argument' [ALSA error -22]
```

This is the most direct mapping of `args` to a formatted string, and will show host_error_message even if it is zero length. It's more consistent than glopping the two error codes into one bracket conditionally. Maybe it's marginally harder to read than having a single listing of error codes at the end of the string, but when you look at it in terms of its components: 

```
pa_error_message [PaErrorCode <xxx>]: host_error_message [<host> error <xxx>]
```

This seems like a more intuitive way of presenting the errors.